donate/payments/views.py

```diff
@@ -329,12 +329,27 @@ def process_monthly_transaction(self, form):
             return self.process_braintree_error_result(result, form)
 
     def get_transaction_details_for_session(self, result, form, **kwargs):
+        card_type = kwargs.get('card_type', 'Unknown')
+
+        if card_type == 'Unknown':
```

Comment:
I would strongly urge against having in here that runs only during testing - this is liable to introduce bugs in production. 

I think the issue is that you're not actually supplying a `card_type` kwarg anywhere, so this kwarg will never be set (see comment above). The mock result classes used in the tests may need to be modified so that they supply this new property in the appropriate places.

Reply:
This function runs fine during normal operation, where this value should be `Unknown` unless the transaction we get from Braintree includes an explicit card_type (it might not, so I want to make doubly sure we have that fallback value set up first).

During normal operation, there is no `card_type` in `kwargs`, ever, as far as I can tell. That only happens during testing, but when it happens during testing, not pulling the value from `kwargs` causes test errors, and this code is complex enough that I literally have no idea _why_ the tests call this function in they way they do.

Can you expand on what other changes would be necessary to the mocked structures? Because I've already added it to the only structure that has `card_type` as far as can be found in Braintree's API documentation (https://github.com/mozilla/donate-wagtail/blob/c60888644da6034efa7252463c97dde9c87a88c7/donate/payments/tests/test_views.py#L39)